dragonfly/block/carpet.go

```diff
@@ -0,0 +1,81 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/colour"
+	"github.com/df-mc/dragonfly/dragonfly/entity/physics"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"time"
+)
+
+// Carpet is a colourful block that can be obtained by killing/shearing sheep, or crafted using four string.
+type Carpet struct {
+	Colour colour.Colour
+}
+
+// AABB ...
+func (Carpet) AABB(world.BlockPos, *world.World) []physics.AABB {
+	return []physics.AABB{physics.NewAABB(mgl64.Vec3{}, mgl64.Vec3{1, 0.0625, 1})}
+}
+
+// BreakInfo ...
+func (c Carpet) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0.1,
+		Harvestable: alwaysHarvestable,
+		Effective:   alwaysEffective,
```

Comment:
This should be `neverEffective`, as no tool is actually effective for breaking a carpet.